jetty-cdi/src/main/java/org/eclipse/jetty/cdi/CdiSpiDecorator.java

```diff
@@ -92,6 +95,35 @@ public CdiSpiDecorator(ServletContextHandler context) throws UnsupportedOperatio
         }
     }
 
+    /**
+     * Test if a class can be decorated.
+     * The default implementation calls {@link #isKnownUndecoratable(String) }
+     * on the class and all it's super classes.
+     * @param clazz The class to check
+     * @return True if the class and all it's super classes can be decorated
+     */
+    protected boolean isDecoratable(Class<?> clazz)
+    {
+        if (Object.class == clazz)
+            return true;
+        if (isKnownUndecoratable(clazz.getName()))
```

Comment:
Not at all fond of the name isKnownDecoratable(). How about instead you remove isKnownDecoratable() and add a field list of _undecoratables, a setter for them, and in isDecoratable() you just check the list? This might also be a good idea in case CDI adds some other classes that can't be decorated?